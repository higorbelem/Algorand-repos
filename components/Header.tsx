import { StyleSheet, TouchableOpacity, View, ViewProps } from "react-native"
import { router } from "expo-router";

import AlgorandLogo from "@/assets/svgs/algorand-logo.svg";
import ChevronLeftIcon from "@/assets/svgs/chevron-left.svg";
import { Colors } from "@/constants/Colors";

import { Text } from "./Text";

export type HeaderProps = ViewProps & {
    showBackButton?: boolean;
    onExplore?: () => void;
};

export const Header = (props: HeaderProps) => {
    return(
        <View style={styles.container}>
            <View style={styles.leftContainer}>
                {
                    props.showBackButton && (
                        <TouchableOpacity testID="back-button" onPress={() => router.back()} activeOpacity={0.7}>
                            <ChevronLeftIcon width={40} height={40}/>
                        </TouchableOpacity>
                    )
                }

                <AlgorandLogo />
            </View>

            {
                props.onExplore && (
                    <TouchableOpacity testID="explore-button" onPress={props.onExplore} activeOpacity={0.7} style={styles.exploreButton}>
                        <Text>Explore</Text>
                    </TouchableOpacity>
                )
            }
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        width: '100%',
        padding: 20
    },
    leftContainer: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    exploreButton: {
        paddingHorizontal: 20,
        borderRadius: 10,
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: Colors.green
    }
});