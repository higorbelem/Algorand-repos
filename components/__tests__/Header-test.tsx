import * as React from 'react';
import { render } from '@testing-library/react-native';

import { Header } from '../Header';

describe('Header component', () => {
  it('Renders correctly', () => {
    const tree = render(<Header/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('Back button', () => {
    it('Should be hidden by default', async () => {
      const tree = render(<Header />);
      
      const component = tree.queryByTestId('back-button');

      expect(component).toBeFalsy();
    });

    it('Should show if "showBackButton" is true', async () => {
      const tree = render(<Header showBackButton/>);
      
      const component = tree.queryByTestId('back-button');

      expect(component).toBeTruthy();
    });
  });

  describe('Explore button', () => {
    it('Should be hidden by default', async () => {
      const tree = render(<Header />);
      
      const component = tree.queryByTestId('explore-button');

      expect(component).toBeFalsy();
    });

    it('Should show if "onExplore" is set', async () => {
      const tree = render(<Header onExplore={() => {}}/>);
      
      const component = tree.queryByTestId('explore-button');

      expect(component).toBeTruthy();
    });
  });
});