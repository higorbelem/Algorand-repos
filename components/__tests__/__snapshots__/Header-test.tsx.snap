// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <svg />
  </View>
</View>
`;
